### PR TITLE
Overhaul of Workshop

### PR DESCRIPTION
**Pull request by asteves**
This PR makes the follows changes:

- Restructures the lesson. The workshop now begins with a discussion of data wrangling principles and the benefits of tidy data. There is also a whole game full example at the beginning to present learners with what the lessons will build to instead of introducing everything piecemeal. 
- Introduction of verbs follow from the discussion of data principles. The ordering of verbs has also been altered. 
- New examples for all verbs other than `pivot_longer()`, `separate()`, and `pivot_wider()`. 
- Many sections with new text. There is a separate Instructor's Notes file for D-Lab instructors, but it is not in this repo by design after discussion with @aculich and @pssachdeva 
- New dataset from the London Fire Brigade about animal rescues, as well as a new synthetic dataset, alongside the usual gapminder dataset. 
- New challenges. Two are conceptual questions about data principles. Six are code exercises. 
- All exercises now have solutions and these solutions are updated to include the latest functions and code patterns. 
- Removal of unneeded files, which also closes a security alert from Dependabot. 
- Standardized file structure to match D-Lab style guide 
- Full update to the README to match D-Lab style guide principles. 

These changes are also all made towards the end goal of this workshop as flowing into the Advanced Data Wrangling workshop.

### CODE REVIEW

**Comment by aculich** [comment]
Great to see the detailed bulleted list of changes for this PR!

**Comment by asteves** [comment]
@aculich I took the basic template from @pssachdeva on the R-Fundamentals repo.

**Review by pssachdeva — COMMENTED** [review]
This is an excellent update, thanks for all the hard work on this. I left some comments in a review of the repository.

A few other things:
- Could you rename the repository R-Data-Wrangling, and all corresponding files to match the title?
- Do you see this being done in one or two workshops? If two, when is the natural end for Part 1? 
- It would be good to fold in the topics from advanced data wrangling into this workshop, and treat it as a Part 3 covering additional topics. that way, we wouldn't have to manage two separate data wrangling workshops. What do you think about this?

File: README.md [hunk 1]
```diff
@@ -1,44 +1,92 @@
 # Data Wrangling and Manipulation in R
 
+## Workshop Goals 
+
 It is often said that 80% of data analysis is spent on the process of cleaning and preparing the data. This workshop will introduce tools (notably dplyr and tidyr) that makes data wrangling and manipulation much easier. Participants will learn how to use these packages to subset and reshape data sets, do calculations across groups of data, clean data, and other useful stuff.
```
[the remark this answers is not in the document]
Comment (asteves) [reply README.md:5]: Done in latest commit.

File: README.md [hunk 2]
```diff
@@ -1,44 +1,92 @@
...
 
-**Prior knowledge**: Previous experience with (basic) R is helpful but not necessary.
```
[the remark this answers is not in the document]
Comment (asteves) [reply README.md:5]: Done in latest commit. Also updated R Fundamentals to replace Fast-R

File: lessons/Tutorial.Rmd [hunk 3]
```diff
@@ -0,0 +1,615 @@
...
+## What is data wrangling 
+
+> It is often said that 80% of data analysis is spent on the process of cleaning and preparing the data (Dasu and Johnson 2003). 
+
+At its heart, data wrangling is the set of techniques to get data into a usable form for analysis and visualization. 
+
+Data wrangling is a catchall for data importing, data cleaning, and data transformation to create some product with the finished data. 
+
+
+## Tidy data
```
[the remark this answers is not in the document]
Comment (asteves) [reply lessons/Tutorial.Rmd:29]: Yes, it's intentional. Related to responses in the general comments, the workshop starts out by providing a framework for thinking about data manipulation and transformation. There's more coverage here in the Instructor Notes for the workshop. 

Happy to discuss more.

File: lessons/Tutorial.Rmd [hunk 4]
```diff
@@ -0,0 +1,615 @@
...
+
+The `here` function is a way to create file paths relative to the top-level directory. 
+
+For a descriptive analysis, we'd like to know the average life expectancy for individuals who are and are not in agriculture for states within each state on the West Coast. To accomplish this task, we will need to filter our list of states, group by each state, create a variable denoting whether the individual works in agriculture or not, and summarize the average age of individuals who do and do not work in agriculture. Finally, we'd like to arrange the states in reverse alphabetical order. 
+
+What follows is an example of the workflow just described in the spirit of [teaching the whole game](https://www.gse.harvard.edu/news/uk/09/01/education-bat-seven-principles-educators)
+
+```{r}
+## Data import 
+jobs <- read_csv(here("data/jobs.csv"))
```
[the remark this answers is not in the document]
Comment (asteves) [reply lessons/Tutorial.Rmd:144]: Done.

File: lessons/Tutorial.Rmd [hunk 5]
```diff
@@ -0,0 +1,615 @@
...
+
+```
+
+## Multiple Filters at once 
+
+Sometimes we want to filter multiple conditions at once. For example, we might be interested in all rescues of birds or horses after 2011. We use the `,` in our filter expression to accomplish this. 
+
+```{r}
+birds_and_horses <- animals %>% 
+  filter(Type == "Bird"|Type == "Horse", Year > 2011)
```
[the remark this answers is not in the document]
Comment (asteves) [reply lessons/Tutorial.Rmd:267]: Done.

File: lessons/Tutorial.Rmd [hunk 6]
```diff
@@ -0,0 +1,615 @@
...
+For a descriptive analysis, we'd like to know the average life expectancy for individuals who are and are not in agriculture for states within each state on the West Coast. To accomplish this task, we will need to filter our list of states, group by each state, create a variable denoting whether the individual works in agriculture or not, and summarize the average age of individuals who do and do not work in agriculture. Finally, we'd like to arrange the states in reverse alphabetical order. 
+
+What follows is an example of the workflow just described in the spirit of [teaching the whole game](https://www.gse.harvard.edu/news/uk/09/01/education-bat-seven-principles-educators)
+
+```{r}
+## Data import 
+jobs <- read_csv(here("data/jobs.csv"))
+
+## Workflow 
+jobs %>%
```
[the remark this answers is not in the document]
Comment (asteves) [reply lessons/Tutorial.Rmd:147]: Done. 

There's also an extended discussion of this workflow with the "English" translation in the Instructor's notes.

File: lessons/Tutorial.Rmd [hunk 7]
```diff
@@ -0,0 +1,615 @@
...
+We often refer to these two opposite structures as "long" vs. "wide" formats. In the "long" format, you usually have 1 column for the observed variable(s), and the other columns are ID variables. 
+
+For the "wide" format, each row is often a site/subject/patient, and you have multiple observation variables containing the same type of data.
+
+## Tidy data 
+
+Let's make the data tidy.
+
+```{r, echo = F}
+challenge_1 %>%
```
[the remark this answers is not in the document]
Comment (asteves) [reply lessons/Tutorial.Rmd:90]: Done. Instructors can also refer to the notes here as well.

File: lessons/Tutorial.Rmd [hunk 8]
```diff
@@ -0,0 +1,615 @@
...
+
+long <- data.frame(
+  name = c("Wilbur", "Petunia", "Gregory", "Wilbur", "Petunia", "Gregory", "Wilbur", "Petunia", "Gregory"),
+  time = c(1, 1, 1, 2, 2, 2, 3, 3, 3),
+  heartrate = c(67, 80, 64, 56, 90, 50, 70, 67, 10)
+)
+long
+```
+
+We often refer to these two opposite structures as "long" vs. "wide" formats. In the "long" format, you usually have 1 column for the observed variable(s), and the other columns are ID variables.
```
[the remark this answers is not in the document]
Comment (asteves) [reply lessons/Tutorial.Rmd]: I think resolved by latest commit. Let me know if not.

**Comment by asteves** [comment]
@pssachdeva 

- Could you rename the repository R-Data-Wrangling, and all corresponding files to match the title?
    - Yes. Will do. 

- Do you see this being done in one or two workshops? If two, when is the natural end for Part 1?
    - My understanding is that this is usually taught as one three hour workshop. I see this as the first part of the wrangling sequence. 
 
- It would be good to fold in the topics from advanced data wrangling into this workshop, and treat it as a Part 3 covering additional topics. that way, we wouldn't have to manage two separate data wrangling workshops. What do you think about this?
    - I *strongly* agree with this. My view is that the "advanced" data wrangling workshop will be parts 2 and 3 of this workshop. Do you think it makes more sense to focus development here entirely? As of now, I've been working on a develop branch in the other repository for changes to the advanced section. 

- This branch is out of date - you can either rebase in order to keep the git history clean (preferable), or just merge the additional changes so that you're up to date.
    - Sure. Is this something I need to do off the command line?

**Pull request merged**